From ce6d952fa15aea5dae34204582fbf1472f01330b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 21 Sep 2016 13:42:59 +0800
Subject: [PATCH] wayland/pointer: Unset pointer focus when disabled

Previously the focus was reset implicitly by a memset() on the whole
MetaWaylandPointer struct. When MetaWaylandPointer was turned into a
GObject, this was not possible any more, and the focus was not updated
properly.

https://bugzilla.gnome.org/show_bug.cgi?id=771646
---
 src/wayland/meta-wayland-pointer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/wayland/meta-wayland-pointer.c b/src/wayland/meta-wayland-pointer.c
index 107b77c..05a8684 100644
--- a/src/wayland/meta-wayland-pointer.c
+++ b/src/wayland/meta-wayland-pointer.c
@@ -494,6 +494,7 @@ meta_wayland_pointer_disable (MetaWaylandPointer *pointer)
     }
 
   meta_wayland_pointer_end_grab (pointer);
+  meta_wayland_pointer_set_focus (pointer, NULL);
 
   g_clear_pointer (&pointer->pointer_clients, g_hash_table_unref);
   pointer->cursor_surface = NULL;
-- 
2.9.3

